qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/QuteCodeActions.java

```diff
@@ -186,15 +199,52 @@ private static void doCodeActionToDisableValidation(Template template, List<Diag
 		codeActions.add(disableValidationForTemplateQuickFix);
 	}
 
+	/**
+	 * Create CodeAction for deprecated `with` Qute syntax.
+	 *
+	 * @param template    the Qute template.
+	 * @param diagnostic  the diagnostic list that this CodeAction will fix.
+	 * @param codeActions the list of CodeActions to perform.
+	 * @throws BadLocationException
+	 *
+	 */
+	private void doCodeActionsForNotRecommendedWithSection(Template template, Diagnostic diagnostic,
+			List<CodeAction> codeActions) {
+		Range withSectionRange = diagnostic.getRange();
+		try {
+			int withSectionStart = template.offsetAt(withSectionRange.getStart());
+			int withSectionEnd = template.offsetAt(withSectionRange.getEnd());
+			WithSection withSection = (WithSection) template.findNodeAt(withSectionStart);
+			String javaResourceClassName = withSection.getObjectParameter().getName();
+			List<Node> withExpressions = withSection.getChildren().stream().filter(s -> s instanceof Expression)
+					.collect(Collectors.toList());
+			List<String> letExpressionParameterList = new ArrayList<String>();
+			for (Node expression : withExpressions) {

```

Comment:
rename expression with child, because the node can be a text, a section, an expression (not only an expression).

Calling just withSection.getChildren() is not enough because #with can include other section which can include other expresion. You need to call again node.#getChildren()

Reply:
![replacewithwithletnested](https://user-images.githubusercontent.com/26389510/156660279-119dd89a-d598-497f-b750-221805639d1b.gif)